### PR TITLE
Introduce AI Scout role

### PR DESCRIPTION
**Pull request by ihhub**
Also increase fog discovery value as it was very low and AI prefers to just stay in one place.

### CODE REVIEW

File: src/fheroes2/ai/normal/ai_normal_kingdom.cpp [hunk 1]
```diff
@@ -131,6 +131,12 @@ namespace
             heroList.pop_back();
         }
 
+        if ( heroList.size() > 2 ) {
+            // Assign the role and remove them so they aren't counted towards the median strength
+            heroList.back().hero->setAIRole( Heroes::Role::SCOUT );
```
Comment (oleg-derevenetz) [inline src/fheroes2/ai/normal/ai_normal_kingdom.cpp:136]: So if we have exactly 3 heroes, we just assign scout role to the first one that came across without taking into account his strength?

**Review by oleg-derevenetz — COMMENTED** [review]
Hi @ihhub I just left one question, please take a look when you have time.

File: src/fheroes2/ai/normal/ai_normal_kingdom.cpp (hunk 1 above)
Comment (ihhub) [reply src/fheroes2/ai/normal/ai_normal_kingdom.cpp:136]: I think this is a bug. We shouldn't do this. I will fix.

Comment (idshibanov) [reply src/fheroes2/ai/normal/ai_normal_kingdom.cpp:136]: You can probably merge this logic with the previous one (for Courier) that sorts the heroes before assigning them.

**Review by oleg-derevenetz — APPROVED** [review]

**Pull request merged**